researchapp/sync_api.py

```diff
@@ -0,0 +1,62 @@
+''' Library for interfacing with the "synchronizer".
+'''
+import requests
+
+
+class SyncExtension(object):
+    ''' Flask extension for interacting with the Synchronizer API.
+    '''
+    def __init__(self, app=None):
+        self.session = None
+        self.host = None
+
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        ''' Init the extension.
+        '''
+        self.session = requests.Session()
+        self.host = app.config['SYNCHRONIZER_HOST']
+
+    def list_providers(self):
+        ''' List available providers.
+        '''
+        resp = self.session.get(self.host + '/providers')
```

Comment:
Are these calls authenticated? If so, where does that happen? If not... I think they should be, at least with a symmetric secret "API Key" header. Of course, we'd also want to put the Research App API behind a firewall, etc...